### PR TITLE
22 create record and record batch types

### PR DESCRIPTION
**Pull request by Kalin8900**

### CODE REVIEW

File: src/protocol/primitives/record-batch-attributes.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+export class Transactional {
+  public static readonly No = 0b00000000;
+  public static readonly Yes = 0b00010000;
+  private static readonly MASK = 0b00000000_00010000;
+
+  public static fromInt16({ value }: Int16): TransactionalValue {
+    const transactional = value & this.MASK;
+
+    if (transactional & Transactional.Yes) {
```
Comment (kordiangrabowski) [inline src/protocol/primitives/record-batch-attributes.ts:55]: Can it be simplified to the `transactional !== 0`?

File: src/protocol/primitives/record-batch-attributes.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+export class DeleteHorizon {
+  public static readonly No = 0b00000000;
+  public static readonly Yes = 0b01000000;
+  private static readonly MASK = 0b00000000_01000000;
+
+  public static fromInt16({ value }: Int16): DeleteHorizonValue {
+    const deleteHorizon = value & this.MASK;
+
+    if (deleteHorizon & DeleteHorizon.Yes) {
```
Comment (kordiangrabowski) [inline src/protocol/primitives/record-batch-attributes.ts:91]: https://github.com/trisquareeu/kafkit/pull/25/files#r1264974215

File: src/protocol/primitives/record-batch-attributes.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+export class TimestampType {
+  public static readonly CreateTime = 0b00000000;
+  public static readonly LogAppendTime = 0b00001000;
+  private static readonly MASK = 0b00000000_00001000;
+
+  public static fromInt16({ value }: Int16): TimestampTypeValue {
+    const timestampType = value & this.MASK;
+
+    if (timestampType & TimestampType.LogAppendTime) {
```
Comment (kordiangrabowski) [inline src/protocol/primitives/record-batch-attributes.ts:37]: https://github.com/trisquareeu/kafkit/pull/25/files#r1264974215

File: src/protocol/primitives/record-batch.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+  public static deserialize(buffer: ReadBuffer): RecordBatch {
+    const baseOffset = Int64.deserialize(buffer);
+    const length = Int32.deserialize(buffer).value;
+    const temporary = new ReadBuffer(buffer.readBuffer(length));
+
+    const partitionLeaderEpoch = Int32.deserialize(temporary);
+    Int8.deserialize(temporary); // magic
+    Int32.deserialize(temporary); // crc
```
Comment (kordiangrabowski) [inline src/protocol/primitives/record-batch.ts:72]: is the checksum ignored?

**Review by kordiangrabowski — CHANGES_REQUESTED** [review]
As the compression is not supported in this PR, consider documenting it by explicitly throwing a relevant error. Please consider verification of the checksum during the deserialization.

File: src/protocol/primitives/record-batch-attributes.ts (hunk 1 above)
Comment (Kalin8900) [reply src/protocol/primitives/record-batch-attributes.ts:55]: Maybe I will just compare it with `Transactional.Yes`?

Comment (kordiangrabowski) [reply src/protocol/primitives/record-batch-attributes.ts:55]: Sounds even better.

Comment (Kalin8900) [reply src/protocol/primitives/record-batch-attributes.ts:55]: Now when I'm thinking about it, after performing AND with MASK, we could just return the value as in Compression. What do you think?

Comment (kordiangrabowski) [reply src/protocol/primitives/record-batch-attributes.ts:55]: The simpler the better 😅

**Review by kordiangrabowski — APPROVED** [review]

**Pull request merged**